src/Node/BlockNode.php

```diff
@@ -37,6 +37,14 @@ public function compile(Compiler $compiler): void
 
         $compiler
             ->subcompile($this->getNode('body'))
+        ;
+
+        if (!$this->getNode('body') instanceof NodeOutputInterface && $compiler->getEnvironment()->useYield()) {
+            // needed when body doesn't yield anything
+            $compiler->write("yield;\n");
```

Comment:
should we yield an empty string instead of `null` to respect the `iterable<string>` return types ?

Reply:
Indeed